src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -252,6 +255,52 @@ private void retryableInferenceSentencesWithSingleVectorResult(
         ));
     }
 
+    /**
+     * Process the highlighting output from ML model response.
+     * Converts the model output into a list of maps containing highlighting information.
+     */
+    private List<Map<String, Object>> processHighlightingOutput(MLOutput mlOutput) {
```

Comment:
Better to accept only `ModelTensorOutput` if that is what we want it to be.

```suggestion
    private List<Map<String, Object>> processHighlightingOutput(ModelTensorOutput mlOutput) {
```

Reply:
ack